main.tf

```diff
@@ -242,6 +242,27 @@ data "aws_iam_policy_document" "this" {
       }
     }
   }
+
+  dynamic "statement" {
+    for_each = length(var.key_service_principals) > 0 ? [1] : []
```

Comment:
this one I think we'll have to handle differently. I think for services, I'm inclined to use just a generic `dynamic "statement"` block because not all services require the full set of API permissions listed below. In addition, with services we want to protect from the confused deputy problem and utilize conditions which means we'll have to iterate over some maps.